app/api/orders.py

```diff
@@ -319,7 +319,7 @@ def before_update_object(self, order, data, view_kwargs):
         :param view_kwargs:
         :return:
         """
-        if data.get('amount') or data.get('is_billing_enabled'):
+        if data.get('amount') or data.get('is_billing_enabled') or order.event.is_billing_info_mandatory:
```

Comment:
@kushthedude You need to see the things at the server level. Yes, an order with `initial` status is created as soon as you "user clicks buy now from the ticket list", but I tried locally and simply made an order with `completed` status in just one API request. We definitely need a check while creating an order as well.